pytest-embedded-jtag/pytest_embedded_jtag/gdb.py

```diff
@@ -1,54 +1,21 @@
+import logging
+import shlex
 from typing import Optional
 
 from pytest_embedded.log import DuplicateStdoutPopen, MessageQueue
 
 
 class Gdb(DuplicateStdoutPopen):
     GDB_PROG_PATH = 'xtensa-esp32-elf-gdb'
-    GDB_DEFAULT_ARGS = '--nx --quiet --interpreter=mi2'
```

Comment:
Will switching mi2 mode off make it harder to get information (like backtraces) in a structured way?

Reply:
mi2 would print a json-like bytestring. since we're using `expect` functions everywhere, normal string output would be more user-friendly IMHO...

if the user need to do it in the mi2 way, could override via CLI option `--gdb-cli-args [ARG]`

Reply:
I think `expect` is fine for unstructured text output (like UART logs), but if we can get the information (like a backtrace) in a structured way via pygdbmi (like we do in IDF panic tests, for instance), I think it would generally be more reliable.

So my suggestion is rather to use mi2 together with pygdbmi, rather than mi2 with expect-like matching.

Reply:
Actually these could be two different ways we set up the DUT, one could use text-mode GDB interactions, the other could use MI2 and pygdbmi.

Reply:
created #143 to do this in a separate PR.